exercises/concept/plane-tickets/.docs/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Instructions
+
+A generator in Python is a _callable function_ that returns a [lazy iterator](https://en.wikipedia.org/wiki/Lazy_evaluation).
+
+_Lazy iterators_ are similar to `lists`, and other `iterators`, but with one key difference: They do not store their `values` in memory, but _generate_ their values when needed.
+
+## Constructing a generator
+
+Constructing a `generator` is a bit different than a normal `function`. You will need the `yield` expression, which we will go into depth with [later](#the-yield-expression).
+
+Lets say you want to construct a `generator` that generates all the _squares_ from a list of numbers. You would construct that function like this:
+
+```python
+>>> def squares(list_of_numbers):
+>>>     for number in list_of_numbers:
+>>>         yield number ** 2
+```
+
```

Comment:
```suggestion
An example is a function that returns the _squares_ from a given list of numbers.  As currently written, all input must be processed before any values can be returned:  

```python
>>> def squares(list_of_numbers):
>>>     squares = []
>>>     for number in list_of_numbers:
>>>         squares.append(number ** 2)
>>>     return squares
```
To convert that function into a _generator_, you would change that function to this:

```python
>>> def squares(list_of_numbers):
>>>     for number in list_of_numbers:
>>>         yield number ** 2
```
```

Reply:
..thinking we might want to add a little rationale here for why you'd do this instead of writing a utility function that did one square at a time.  Not sure (yet) what I would add though.